File: web/src/app/shared/models/attendee.ts
export class TrekAttendee {
    isAdult: boolean;
    firstName: string;
    lastName: string;
    ward: string;
    gender: string;
    peanutAllergy: boolean;
    glutenAllergy: boolean;
    dateOfBirth: Date;
    parentName: string;
    parentEmail: string;
    parentPhone: string;
    emergencyName: string;
    emergencyPhone: string;
  
    constructor(obj: TrekAttendee = {} as TrekAttendee) {
        this.isAdult = obj.isAdult;
        this.firstName = obj.firstName;
        this.lastName = obj.lastName;
        this.ward = obj.ward
        this.gender = obj.gender
        this.peanutAllergy = obj.peanutAllergy;
        this.glutenAllergy = obj.glutenAllergy;
        this.dateOfBirth = obj.dateOfBirth;
        this.parentName = obj.parentName;
        this.parentEmail = obj.parentEmail;
        this.parentPhone = obj.parentPhone;
        this.emergencyName = obj.emergencyName;
        this.emergencyPhone = obj.emergencyPhone;
    }
  }
  

File: api/trek.api/Models/Entities/Attendee.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace trek.api.Models.Entities
{
    public class Attendee
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public bool IsAdult { get; set; }

        [Required]
        [MaxLength(100)]
        public string FirstName { get; set; }

        [Required]
        [MaxLength(100)]
        public string LastName { get; set; }

        [Required]
        [MaxLength(50)]
        public string Ward { get; set; }

        [Required]
        [MaxLength(10)]
        public string Gender { get; set; }

        public bool PeanutAllergy { get; set; }

        public bool GlutenAllergy { get; set; }

        public DateTime? DateOfBirth { get; set; }

        [MaxLength(100)]
        public string ParentName { get; set; }

        [MaxLength(100)]
        public string ParentEmail { get; set; }

        [MaxLength(15)]
        public string ParentPhone { get; set; }

        [Required]
        [MaxLength(100)]
        public string EmergencyName { get; set; }

        [Required]
        [MaxLength(15)]
        public string EmergencyPhone { get; set; }

        [Required]
        public DateTime RegisteredDateTime { get; set; }
    }
}
